client/app/telecom/telephony/alias/configuration/changeType/telecom-telephony-alias-configuration-changeType.controller.js

```diff
@@ -1,170 +1,214 @@
-        { id: 'contactCenterSolution', label: $translate.instant('telephony_alias_config_change_type_label_contactCenterSolution') },
-        { id: 'contactCenterSolutionExpert', label: $translate.instant('telephony_alias_config_change_type_label_contactCenterSolutionExpert') },
-        { id: 'empty', label: $translate.instant('telephony_alias_config_change_type_label_empty') },
-      ];
-
-      self.number.feature.startEdition();
-      self.noCache = false;
-
-      return self.number;
-    }).catch((error) => {
-      Toast.error([$translate.instant('telephony_alias_load_error'), (error.data && error.data.message) || ''].join(' '));
-      return $q.reject(error);
-    })
-      .finally(() => {
-        self.loading.line = false;
-      });
+
+    this.simpleModeOptions = ['cloudIvr', 'conference', 'contactCenterSolution', 'redirect'];
+    this.expertModeOptions = ['contactCenterSolutionExpert', 'svi'];
   }
 
-  $scope.$on('$destroy', () => {
-    if (self.number && self.number.feature) {
-      self.number.feature.stopEdition(true);
-    }
-  });
+  $onInit() {
+    this.number = null;
+    this.loading = true;
+
+    this.bulkData = {
+      infos: {
+        name: 'configurationNumberChangeType',
+        actions: [{
+          name: 'options',
+          route: '/telephony/{billingAccount}/number/{serviceName}/changeFeatureType',
+          method: 'POST',
+          params: null,
+        }],
+      },
+    };
 
-  /* -----  End of INITIALIZATION  ------*/
+    this.voipService.fetchSingleService(this.billingAccount, this.serviceName).then((number) => {
+      this.number = number;
+      this.initSelectedFeatureType();
 
-  /* ===========================
-  =            BULK            =
-  ============================ */
+      return this.number;
+    }).catch((error) => {
+      this.Toast.error(
+        `${this.$translate.instant('telephony_alias_load_error')} ${_.get(error, 'data.message', '')}`,
+      );
+    }).finally(() => {
+      this.loading = false;
+    });
+  }
 
-  self.bulkDatas = {
-    infos: {
-      name: 'configurationNumberChangeType',
-      actions: [{
-        name: 'options',
-        route: '/telephony/{billingAccount}/number/{serviceName}/changeFeatureType',
-        method: 'POST',
-        params: null,
-      }],
-    },
-  };
-
-  self.getBulkParams = function () {
-    const data = {
-      featureType: self.number.feature.featureType,
+  initSelectedFeatureType() {
+    this.selectedFeatureType = {
+      id: this.number.featureType,
+      name: this.$translate.instant(`telephony_alias_config_change_type_desc_${this.number.featureType}`),
     };
+  }
 
-    return data;
-  };
+  canChangeType() {
+    if (this.showExpertOptions) {
+      return this.expertModeOptions.includes(this.selectedFeatureType.id)
+       && !_(this.selectedFeatureType.id).isEqual(this.number.featureType);
```

Comment:
you could directly use `===` if those are not objects 
also native `includes` is not supported by `IE11`

Reply:
Yes, there is no polyfill in telecom environment

Reply:
Yes :)